ocaml/idl/datamodel_cluster.ml

```diff
@@ -80,6 +80,15 @@ let create =
          }
        ]
       @ timeout_params
+      @ [
+          {
+            param_type= Map (Int, Ref _pif)
```

Comment:
This adds a new reference to another object, do we need to declare this dependency elsewhere in the datamodel? There is a place where you can declare 1:1, and 1-to-many relationships, so that the DB GC properly updates things.
E.g. what if you destroy a PIF, should it be automatically removed from here, do we need to take any action on the cluster side before we do that?

Normally we require PIFs to be marked with disallow-unplug. Obviously we can't require all PIFs to be marked like that all the time, so can we manage these somehow dynamically?

Reply:
I agree, what you are suggesting is the ability to dynamically configure the pifs in a cluster through apis that operate on the db, which will then be propagated to clusterd. We probably need a new api for this, and intergrate it with existing pif destroying calls, etc.